Added documentation for helping users encountering locale error during installation

sudheesh001 opened the pull request.

Osmose commented:
Thanks for the patch! You're getting this issue during the peep install during setup? Do you have an error message you can share with more info?

I'm holding on merging this until I understand what problem this fix is solving in the first place, although maybe I missed a bug that's already been filed for this?

sudheesh001 commented:
Yes @Osmose I am getting this during the peep installation, as far as i remember it was a value error telling locale is not set, if I remember correctly it was something like this
`raise ValueError, 'unknown locale: %s' % localename`
So this was fixed by adding locale as follows
```sh
export LC_ALL=en_US.UTF-8
export LANG=en_US.UTF-8
```

jotes commented:
Hi, could you post full stacktrace?

sudheesh001 commented:
Here I could recreate it on a test environment
```
$ ./bin/peep.py install -r requirements.txt
Downloading sphinx-rtd-theme==0.1.8 (418K)...
Downloading Sphinx==1.3.1 (1334K)...
Downloading graphviz==0.4.7 (12K)...
Downloading argparse==1.3.0 (22K)...
Downloading certifi==14.05.14 (168K)...
Downloading chardet==2.3.0 (164K)...
Downloading commonware==0.4.3 (7K)...
Downloading configparser==3.5.0b2 (38K)...
Downloading diff-match-patch==20121119 (54K)...
Downloading django-session-csrf==0.5 (8K)...
Downloading django-sha2==0.4 (8K)...
Downloading docutils==0.12 (1618K)...
Downloading mock==1.3.0 (56K)...
Downloading pbr==1.6.0 (87K)...
Downloading funcsigs==0.4 (14K)...
Downloading oauthlib==0.1.3 (24K)...
Downloading polib==1.0.6 (24K)...
Downloading py-bcrypt==0.3 (17K)...
Downloading pyasn1==0.1.7 (68K)...
Downloading python-l20n (32K)...
Downloading python-Levenshtein==0.11.2 (45K)...
Downloading pytz==2015.2 (476K)...
Downloading requests==2.6.2 (470K)...
Downloading rsa==3.1.4 (36K)...
Downloading silme==0.9.2...
Downloading six==1.9.0 (10K)...
Downloading suds==0.4 (104K)...
Downloading test-utils (18K)...
Downloading translate-toolkit==1.13.0 (4624K)...
Downloading dj-database-url==0.3.0 (4K)...
Downloading django-dotenv==1.3.0 (3K)...
Downloading django==1.8.7 (6169K)...
Downloading django-jinja==1.3.3 (14K)...
Downloading django-pipeline==1.5.3 (28K)...
Downloading gunicorn==19.3.0 (110K)...
Downloading psycopg2==2.6 (367K)...
Downloading whitenoise==1.0.6 (15K)...
Downloading newrelic==2.50.0.39 (662K)...
Downloading django-browserid==1.0.0 (100K)...
Downloading factory-boy==2.5.2 (33K)...
Downloading nose==1.3.6 (154K)...
Downloading django-nose==1.4 (32K)...
Downloading blessings==1.6 (19K)...
Downloading nose-progressive==1.5.1 (37K)...
Downloading raygun4py (33K)...
Downloading jsonpickle==0.7.0 (34K)...
Downloading blinker==1.3 (91K)...
Downloading lxml==3.4.4 (3537K)...
Downloading mercurial==3.4.1 (4294K)...
Downloading django-sslify==0.2.7 (127K)...
Downloading snowballstemmer==1.2.0 (49K)...
Downloading babel==1.3 (3401K)...
Downloading Jinja2==2.5.5 (438K)...
Downloading Pygments==2.0.2 (672K)...
Downloading alabaster==0.7.4 (21K)...
Downloading django-bulk-update==1.1.4 (13K)...
Downloading parsimonious==0.6.2 (35K)...
Downloading jsonfield==1.0.3 (7K)...
Downloading pylibmc==1.5.0 (53K)...
Downloading django-pylibmc==0.6.0 (8K)...
Downloading django-dirtyfields==0.7 (4K)...
Downloading py-dateutil==2.2 (64K)...
Downloading celery==3.1.18 (515K)...
Downloading billiard==3.3.0.20 (150K)...
Downloading kombu==3.0.26 (238K)...
Downloading amqp==1.4.6 (49K)...
Downloading anyjson==0.3.3 (8K)...
Downloading pylama==6.4.0 (76K)...
Downloading django-guardian==1.3 (292K)...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-fs7tGa/sphinx_rtd_theme-0.1.8-py2.py3-none-any.whl
Installing collected packages: sphinx-rtd-theme
Successfully installed sphinx-rtd-theme
Cleaning up...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-lc8eEp/Sphinx-1.3.1-py2.py3-none-any.whl
Installing collected packages: Sphinx
Successfully installed Sphinx
Cleaning up...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-_S_2dY/graphviz-0.4.7-py2.py3-none-any.whl
Installing collected packages: graphviz
Successfully installed graphviz
Cleaning up...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-SaBLUj/argparse-1.3.0-py2.py3-none-any.whl
Installing collected packages: argparse
Successfully installed argparse
Cleaning up...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-hzvm2F/certifi-14.05.14.tar.gz
  Running setup.py (path:/var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/pip-NdHnfC-build/setup.py) egg_info for package from file:///var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-hzvm2F/certifi-14.05.14.tar.gz

Installing collected packages: certifi
  Running setup.py install for certifi

Successfully installed certifi
Cleaning up...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-IDj_on/chardet-2.3.0.tar.gz
  Running setup.py (path:/var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/pip-5eLI4k-build/setup.py) egg_info for package from file:///var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-IDj_on/chardet-2.3.0.tar.gz

    warning: no files found matching 'COPYING'
    warning: no files found matching '*.html' under directory 'docs'
    warning: no files found matching '*.css' under directory 'docs'
    warning: no files found matching '*.png' under directory 'docs'
    warning: no files found matching '*.gif' under directory 'docs'
Installing collected packages: chardet
  Running setup.py install for chardet

    warning: no files found matching 'COPYING'
    warning: no files found matching '*.html' under directory 'docs'
    warning: no files found matching '*.css' under directory 'docs'
    warning: no files found matching '*.png' under directory 'docs'
    warning: no files found matching '*.gif' under directory 'docs'
    Installing chardetect script to /Users/sudheeshsinganamalla/Documents/pontoon/testvenv/bin
Successfully installed chardet
Cleaning up...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-QAO8Ao/commonware-0.4.3.tar.gz
  Running setup.py (path:/var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/pip-eOAAkW-build/setup.py) egg_info for package from file:///var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-QAO8Ao/commonware-0.4.3.tar.gz

Installing collected packages: commonware
  Running setup.py install for commonware

Successfully installed commonware
Cleaning up...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-Nnlzt8/configparser-3.5.0b2.tar.gz
  Running setup.py (path:/var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/pip-ZAg2Fy-build/setup.py) egg_info for package from file:///var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-Nnlzt8/configparser-3.5.0b2.tar.gz

Installing collected packages: configparser
  Running setup.py install for configparser

    Skipping installation of /Users/sudheeshsinganamalla/Documents/pontoon/testvenv/lib/python2.7/site-packages/backports/__init__.py (namespace package)
    Installing /Users/sudheeshsinganamalla/Documents/pontoon/testvenv/lib/python2.7/site-packages/configparser-3.5.0b2-py2.7-nspkg.pth
Successfully installed configparser
Cleaning up...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-rf_f5L/diff-match-patch-20121119.tar.gz
  Running setup.py (path:/var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/pip-cO9d96-build/setup.py) egg_info for package from file:///var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-rf_f5L/diff-match-patch-20121119.tar.gz

Installing collected packages: diff-match-patch
  Running setup.py install for diff-match-patch

Successfully installed diff-match-patch
Cleaning up...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-r3GowW/django-session-csrf-0.5.tar.gz
  Running setup.py (path:/var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/pip-nNwogF-build/setup.py) egg_info for package from file:///var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-r3GowW/django-session-csrf-0.5.tar.gz

Installing collected packages: django-session-csrf
  Running setup.py install for django-session-csrf

Successfully installed django-session-csrf
Cleaning up...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-_oY4vW/django-sha2-0.4.tar.gz
  Running setup.py (path:/var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/pip-8T9RrF-build/setup.py) egg_info for package from file:///var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-_oY4vW/django-sha2-0.4.tar.gz

Installing collected packages: django-sha2
  Running setup.py install for django-sha2

Successfully installed django-sha2
Cleaning up...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-ycXGbv/docutils-0.12.tar.gz
  Running setup.py (path:/var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/pip-AuPWSZ-build/setup.py) egg_info for package from file:///var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-ycXGbv/docutils-0.12.tar.gz

    warning: no files found matching 'MANIFEST'
    warning: no files found matching '*' under directory 'extras'
    warning: no previously-included files matching '.cvsignore' found under directory '*'
    warning: no previously-included files matching '*.pyc' found under directory '*'
    warning: no previously-included files matching '*~' found under directory '*'
    warning: no previously-included files matching '.DS_Store' found under directory '*'
Installing collected packages: docutils
  Running setup.py install for docutils
    changing mode of build/scripts-2.7/rst2html.py from 644 to 755
    changing mode of build/scripts-2.7/rst2s5.py from 644 to 755
    changing mode of build/scripts-2.7/rst2latex.py from 644 to 755
    changing mode of build/scripts-2.7/rst2xetex.py from 644 to 755
    changing mode of build/scripts-2.7/rst2man.py from 644 to 755
    changing mode of build/scripts-2.7/rst2xml.py from 644 to 755
    changing mode of build/scripts-2.7/rst2pseudoxml.py from 644 to 755
    changing mode of build/scripts-2.7/rstpep2html.py from 644 to 755
    changing mode of build/scripts-2.7/rst2odt.py from 644 to 755
    changing mode of build/scripts-2.7/rst2odt_prepstyles.py from 644 to 755

    warning: no files found matching 'MANIFEST'
    warning: no files found matching '*' under directory 'extras'
    warning: no previously-included files matching '.cvsignore' found under directory '*'
    warning: no previously-included files matching '*.pyc' found under directory '*'
    warning: no previously-included files matching '*~' found under directory '*'
    warning: no previously-included files matching '.DS_Store' found under directory '*'
    changing mode of /Users/sudheeshsinganamalla/Documents/pontoon/testvenv/bin/rst2html.py to 755
    changing mode of /Users/sudheeshsinganamalla/Documents/pontoon/testvenv/bin/rst2latex.py to 755
    changing mode of /Users/sudheeshsinganamalla/Documents/pontoon/testvenv/bin/rst2man.py to 755
    changing mode of /Users/sudheeshsinganamalla/Documents/pontoon/testvenv/bin/rst2odt.py to 755
    changing mode of /Users/sudheeshsinganamalla/Documents/pontoon/testvenv/bin/rst2odt_prepstyles.py to 755
    changing mode of /Users/sudheeshsinganamalla/Documents/pontoon/testvenv/bin/rst2pseudoxml.py to 755
    changing mode of /Users/sudheeshsinganamalla/Documents/pontoon/testvenv/bin/rst2s5.py to 755
    changing mode of /Users/sudheeshsinganamalla/Documents/pontoon/testvenv/bin/rst2xetex.py to 755
    changing mode of /Users/sudheeshsinganamalla/Documents/pontoon/testvenv/bin/rst2xml.py to 755
    changing mode of /Users/sudheeshsinganamalla/Documents/pontoon/testvenv/bin/rstpep2html.py to 755
Successfully installed docutils
Cleaning up...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-lNMtXQ/mock-1.3.0-py2.py3-none-any.whl
Installing collected packages: mock
Successfully installed mock
Cleaning up...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-q8naiT/pbr-1.6.0-py2.py3-none-any.whl
Installing collected packages: pbr
Successfully installed pbr
Cleaning up...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-MIMKXO/funcsigs-0.4-py2.py3-none-any.whl
Installing collected packages: funcsigs
Successfully installed funcsigs
Cleaning up...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-IGewUC/oauthlib-0.1.3.tar.gz
  Running setup.py (path:/var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/pip-Uewi5y-build/setup.py) egg_info for package from file:///var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-IGewUC/oauthlib-0.1.3.tar.gz

Installing collected packages: oauthlib
  Running setup.py install for oauthlib

Successfully installed oauthlib
Cleaning up...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-gcVV6E/polib-1.0.6-py2.py3-none-any.whl
Installing collected packages: polib
Successfully installed polib
Cleaning up...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-B8BVA3/py-bcrypt-0.3.tar.gz
  Running setup.py (path:/var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/pip-iuRfW1-build/setup.py) egg_info for package from file:///var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-B8BVA3/py-bcrypt-0.3.tar.gz

Installing collected packages: py-bcrypt
  Running setup.py install for py-bcrypt
    building 'bcrypt._bcrypt' extension
    clang -fno-strict-aliasing -fno-common -dynamic -I/usr/local/include -I/usr/local/opt/sqlite/include -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/include/python2.7 -c bcrypt/bcrypt_python.c -o build/temp.macosx-10.10-x86_64-2.7/bcrypt/bcrypt_python.o
    bcrypt/bcrypt_python.c:56:19: warning: passing 'char *' to parameter of type 'u_int8_t *' (aka 'unsigned char *') converts between pointers to integer types with different sign [-Wpointer-sign]
            encode_salt(ret, csalt, csaltlen, log_rounds);
                             ^~~~~
    bcrypt/bcrypt_python.c:29:36: note: passing argument to parameter here
    void encode_salt(char *, u_int8_t *, u_int16_t, u_int8_t);
                                       ^
    1 warning generated.
    clang -fno-strict-aliasing -fno-common -dynamic -I/usr/local/include -I/usr/local/opt/sqlite/include -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/include/python2.7 -c bcrypt/blowfish.c -o build/temp.macosx-10.10-x86_64-2.7/bcrypt/blowfish.o
    clang -fno-strict-aliasing -fno-common -dynamic -I/usr/local/include -I/usr/local/opt/sqlite/include -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/include/python2.7 -c bcrypt/bcrypt.c -o build/temp.macosx-10.10-x86_64-2.7/bcrypt/bcrypt.o
    clang -bundle -undefined dynamic_lookup -L/usr/local/lib -L/usr/local/opt/sqlite/lib build/temp.macosx-10.10-x86_64-2.7/bcrypt/bcrypt_python.o build/temp.macosx-10.10-x86_64-2.7/bcrypt/blowfish.o build/temp.macosx-10.10-x86_64-2.7/bcrypt/bcrypt.o -o build/lib.macosx-10.10-x86_64-2.7/bcrypt/_bcrypt.so

Successfully installed py-bcrypt
Cleaning up...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-jffR1H/pyasn1-0.1.7.tar.gz
  Running setup.py (path:/var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/pip-8bXZCE-build/setup.py) egg_info for package from file:///var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-jffR1H/pyasn1-0.1.7.tar.gz

Installing collected packages: pyasn1
  Running setup.py install for pyasn1

Successfully installed pyasn1
Cleaning up...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-kg3yZL/python-l20n-4e32a8da09ad7e0013a8a56c3313545eed577d3d.zip
  Running setup.py (path:/var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/pip-XAEg4S-build/setup.py) egg_info for package from file:///var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-kg3yZL/python-l20n-4e32a8da09ad7e0013a8a56c3313545eed577d3d.zip

Installing collected packages: l20n
  Running setup.py install for l20n

Successfully installed l20n
Cleaning up...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-xNEYJj/python-Levenshtein-0.11.2.tar.gz
  Running setup.py (path:/var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/pip-pHMr_y-build/setup.py) egg_info for package from file:///var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-xNEYJj/python-Levenshtein-0.11.2.tar.gz

    warning: no files found matching '*' under directory 'docs'
    warning: no previously-included files matching '*pyc' found anywhere in distribution
    warning: no previously-included files matching '.project' found anywhere in distribution
    warning: no previously-included files matching '.pydevproject' found anywhere in distribution
Installing collected packages: python-Levenshtein
  Running setup.py install for python-Levenshtein
    building 'Levenshtein' extension
    clang -fno-strict-aliasing -fno-common -dynamic -I/usr/local/include -I/usr/local/opt/sqlite/include -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/include/python2.7 -c Levenshtein.c -o build/temp.macosx-10.10-x86_64-2.7/Levenshtein.o
    Levenshtein.c:694:13: warning: assigning to 'lev_byte *' (aka 'unsigned char *') from 'char [1]' converts between pointers to integer types with different sign [-Wpointer-sign]
        string1 = PyString_AS_STRING(arg1);
                ^ ~~~~~~~~~~~~~~~~~~~~~~~~
    Levenshtein.c:695:13: warning: assigning to 'lev_byte *' (aka 'unsigned char *') from 'char [1]' converts between pointers to integer types with different sign [-Wpointer-sign]
        string2 = PyString_AS_STRING(arg2);
                ^ ~~~~~~~~~~~~~~~~~~~~~~~~
    Levenshtein.c:779:13: warning: assigning to 'lev_byte *' (aka 'unsigned char *') from 'char [1]' converts between pointers to integer types with different sign [-Wpointer-sign]
        string1 = PyString_AS_STRING(arg1);
                ^ ~~~~~~~~~~~~~~~~~~~~~~~~
    Levenshtein.c:780:13: warning: assigning to 'lev_byte *' (aka 'unsigned char *') from 'char [1]' converts between pointers to integer types with different sign [-Wpointer-sign]
        string2 = PyString_AS_STRING(arg2);
                ^ ~~~~~~~~~~~~~~~~~~~~~~~~
    Levenshtein.c:823:13: warning: assigning to 'lev_byte *' (aka 'unsigned char *') from 'char [1]' converts between pointers to integer types with different sign [-Wpointer-sign]
        string1 = PyString_AS_STRING(arg1);
                ^ ~~~~~~~~~~~~~~~~~~~~~~~~
    Levenshtein.c:824:13: warning: assigning to 'lev_byte *' (aka 'unsigned char *') from 'char [1]' converts between pointers to integer types with different sign [-Wpointer-sign]
        string2 = PyString_AS_STRING(arg2);
                ^ ~~~~~~~~~~~~~~~~~~~~~~~~
    Levenshtein.c:873:13: warning: assigning to 'lev_byte *' (aka 'unsigned char *') from 'char [1]' converts between pointers to integer types with different sign [-Wpointer-sign]
        string1 = PyString_AS_STRING(arg1);
                ^ ~~~~~~~~~~~~~~~~~~~~~~~~
    Levenshtein.c:874:13: warning: assigning to 'lev_byte *' (aka 'unsigned char *') from 'char [1]' converts between pointers to integer types with different sign [-Wpointer-sign]
        string2 = PyString_AS_STRING(arg2);
                ^ ~~~~~~~~~~~~~~~~~~~~~~~~
    Levenshtein.c:975:43: warning: passing 'lev_byte *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
          result = PyString_FromStringAndSize(medstr, len);
                                              ^~~~~~
    /usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/include/python2.7/stringobject.h:62:63: note: passing argument to parameter here
    PyAPI_FUNC(PyObject *) PyString_FromStringAndSize(const char *, Py_ssize_t);
                                                                  ^
    Levenshtein.c:1054:15: warning: initializing 'lev_byte *' (aka 'unsigned char *') with an expression of type 'char [1]' converts between pointers to integer types with different sign [-Wpointer-sign]
        lev_byte *s = PyString_AS_STRING(arg1);
                  ^   ~~~~~~~~~~~~~~~~~~~~~~~~
    Levenshtein.c:1060:43: warning: passing 'lev_byte *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
          result = PyString_FromStringAndSize(medstr, len);
                                              ^~~~~~
    /usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/include/python2.7/stringobject.h:62:63: note: passing argument to parameter here
    PyAPI_FUNC(PyObject *) PyString_FromStringAndSize(const char *, Py_ssize_t);
                                                                  ^
    Levenshtein.c:1101:26: warning: format specifies type 'int' but the argument has type 'size_t' (aka 'unsigned long') [-Wformat]
                       name, n, PyList_GET_SIZE(wlist));
                             ^
    Levenshtein.c:1101:29: warning: format specifies type 'int' but the argument has type 'Py_ssize_t' (aka 'long') [-Wformat]
                       name, n, PyList_GET_SIZE(wlist));
                                ^~~~~~~~~~~~~~~~~~~~~~
    /usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/include/python2.7/listobject.h:63:32: note: expanded from macro 'PyList_GET_SIZE'
    #define PyList_GET_SIZE(op)    Py_SIZE(op)
                                   ^~~~~~~~~~~
    /usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/include/python2.7/object.h:116:33: note: expanded from macro 'Py_SIZE'
    #define Py_SIZE(ob)             (((PyVarObject*)(ob))->ob_size)
                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Levenshtein.c:1115:61: warning: format specifies type 'int' but the argument has type 'size_t' (aka 'unsigned long') [-Wformat]
                         "%s weight #%i is not a Number", name, i);
                                     ~~                         ^
                                     %zu
    Levenshtein.c:1124:57: warning: format specifies type 'int' but the argument has type 'size_t' (aka 'unsigned long') [-Wformat]
                         "%s weight #%i is negative", name, i);
                                     ~~                     ^
                                     %zu
    Levenshtein.c:1184:16: warning: assigning to 'lev_byte *' (aka 'unsigned char *') from 'char [1]' converts between pointers to integer types with different sign [-Wpointer-sign]
        strings[0] = PyString_AS_STRING(first);
                   ^ ~~~~~~~~~~~~~~~~~~~~~~~~~
    Levenshtein.c:1193:59: warning: format specifies type 'int' but the argument has type 'size_t' (aka 'unsigned long') [-Wformat]
                         "%s item #%i is not a String", name, i);
                                   ~~                         ^
                                   %zu
    Levenshtein.c:1196:18: warning: assigning to 'lev_byte *' (aka 'unsigned char *') from 'char [1]' converts between pointers to integer types with different sign [-Wpointer-sign]
          strings[i] = PyString_AS_STRING(item);
                     ^ ~~~~~~~~~~~~~~~~~~~~~~~~
    Levenshtein.c:1229:60: warning: format specifies type 'int' but the argument has type 'size_t' (aka 'unsigned long') [-Wformat]
                         "%s item #%i is not a Unicode", name, i);
                                   ~~                          ^
                                   %zu
    Levenshtein.c:1613:13: warning: assigning to 'lev_byte *' (aka 'unsigned char *') from 'char [1]' converts between pointers to integer types with different sign [-Wpointer-sign]
        string1 = PyString_AS_STRING(arg1);
                ^ ~~~~~~~~~~~~~~~~~~~~~~~~
    Levenshtein.c:1614:13: warning: assigning to 'lev_byte *' (aka 'unsigned char *') from 'char [1]' converts between pointers to integer types with different sign [-Wpointer-sign]
        string2 = PyString_AS_STRING(arg2);
                ^ ~~~~~~~~~~~~~~~~~~~~~~~~
    Levenshtein.c:1731:13: warning: assigning to 'lev_byte *' (aka 'unsigned char *') from 'char [1]' converts between pointers to integer types with different sign [-Wpointer-sign]
        string1 = PyString_AS_STRING(arg1);
                ^ ~~~~~~~~~~~~~~~~~~~~~~~~
    Levenshtein.c:1732:13: warning: assigning to 'lev_byte *' (aka 'unsigned char *') from 'char [1]' converts between pointers to integer types with different sign [-Wpointer-sign]
        string2 = PyString_AS_STRING(arg2);
                ^ ~~~~~~~~~~~~~~~~~~~~~~~~
    Levenshtein.c:1826:13: warning: assigning to 'lev_byte *' (aka 'unsigned char *') from 'char [1]' converts between pointers to integer types with different sign [-Wpointer-sign]
        string1 = PyString_AS_STRING(arg1);
                ^ ~~~~~~~~~~~~~~~~~~~~~~~~
    Levenshtein.c:1827:13: warning: assigning to 'lev_byte *' (aka 'unsigned char *') from 'char [1]' converts between pointers to integer types with different sign [-Wpointer-sign]
        string2 = PyString_AS_STRING(arg2);
                ^ ~~~~~~~~~~~~~~~~~~~~~~~~
    Levenshtein.c:1841:43: warning: passing 'lev_byte *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
          result = PyString_FromStringAndSize(s, len);
                                              ^
    /usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/include/python2.7/stringobject.h:62:63: note: passing argument to parameter here
    PyAPI_FUNC(PyObject *) PyString_FromStringAndSize(const char *, Py_ssize_t);
                                                                  ^
    Levenshtein.c:1857:43: warning: passing 'lev_byte *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
          result = PyString_FromStringAndSize(s, len);
                                              ^
    /usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/include/python2.7/stringobject.h:62:63: note: passing argument to parameter here
    PyAPI_FUNC(PyObject *) PyString_FromStringAndSize(const char *, Py_ssize_t);
                                                                  ^
    Levenshtein.c:6586:1: warning: unused function 'lev_editops_total_cost' [-Wunused-function]
    lev_editops_total_cost(size_t n,
    ^
    Levenshtein.c:6656:1: warning: unused function 'lev_opcodes_total_cost' [-Wunused-function]
    lev_opcodes_total_cost(size_t nb,
    ^
    Levenshtein.c:6611:1: warning: unused function 'lev_editops_normalize' [-Wunused-function]
    lev_editops_normalize(size_t n,
    ^
    Levenshtein.c:2327:1: warning: unused function 'lev_edit_distance_sod' [-Wunused-function]
    lev_edit_distance_sod(size_t len, const lev_byte *string,
    ^
    Levenshtein.c:2506:1: warning: unused function 'lev_u_edit_distance_sod' [-Wunused-function]
    lev_u_edit_distance_sod(size_t len, const lev_wchar *string,
    ^
    32 warnings generated.
    clang -bundle -undefined dynamic_lookup -L/usr/local/lib -L/usr/local/opt/sqlite/lib build/temp.macosx-10.10-x86_64-2.7/Levenshtein.o -o build/lib.macosx-10.10-x86_64-2.7/Levenshtein.so

    warning: no files found matching '*' under directory 'docs'
    warning: no previously-included files matching '*pyc' found anywhere in distribution
    warning: no previously-included files matching '.project' found anywhere in distribution
    warning: no previously-included files matching '.pydevproject' found anywhere in distribution
Successfully installed python-Levenshtein
Cleaning up...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-yrxmea/pytz-2015.2-py2.py3-none-any.whl
Installing collected packages: pytz
Successfully installed pytz
Cleaning up...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-oUYfqM/requests-2.6.2-py2.py3-none-any.whl
Installing collected packages: requests
Successfully installed requests
Cleaning up...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-Wg5tom/rsa-3.1.4.tar.gz
  Running setup.py (path:/var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/pip-QE7xu6-build/setup.py) egg_info for package from file:///var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-Wg5tom/rsa-3.1.4.tar.gz

    warning: no files found matching 'README'
Installing collected packages: rsa
  Running setup.py install for rsa

    warning: no files found matching 'README'
    Installing pyrsa-encrypt-bigfile script to /Users/sudheeshsinganamalla/Documents/pontoon/testvenv/bin
    Installing pyrsa-encrypt script to /Users/sudheeshsinganamalla/Documents/pontoon/testvenv/bin
    Installing pyrsa-verify script to /Users/sudheeshsinganamalla/Documents/pontoon/testvenv/bin
    Installing pyrsa-sign script to /Users/sudheeshsinganamalla/Documents/pontoon/testvenv/bin
    Installing pyrsa-priv2pub script to /Users/sudheeshsinganamalla/Documents/pontoon/testvenv/bin
    Installing pyrsa-decrypt script to /Users/sudheeshsinganamalla/Documents/pontoon/testvenv/bin
    Installing pyrsa-decrypt-bigfile script to /Users/sudheeshsinganamalla/Documents/pontoon/testvenv/bin
    Installing pyrsa-keygen script to /Users/sudheeshsinganamalla/Documents/pontoon/testvenv/bin
Successfully installed rsa
Cleaning up...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-k96bOa/silme-0.9.2.zip
  Running setup.py (path:/var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/pip-ulhd6c-build/setup.py) egg_info for package from file:///var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-k96bOa/silme-0.9.2.zip

Installing collected packages: silme
  Running setup.py install for silme

Successfully installed silme
Cleaning up...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-QGTWJ1/six-1.9.0-py2.py3-none-any.whl
Installing collected packages: six
Successfully installed six
Cleaning up...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-RIXstq/suds-0.4.tar.gz
  Running setup.py (path:/var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/pip-pvXyU1-build/setup.py) egg_info for package from file:///var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-RIXstq/suds-0.4.tar.gz

Installing collected packages: suds
  Running setup.py install for suds
    /Users/sudheeshsinganamalla/Documents/pontoon/testvenv/bin/python -O /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/tmpRcpqdu.py
    removing /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/tmpRcpqdu.py

Successfully installed suds
Cleaning up...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-ZQHmYR/test-utils-e42e031d31ea63d593e47d564545149a0f30b95f.zip
  Running setup.py (path:/var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/pip-trmSyT-build/setup.py) egg_info for package from file:///var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-ZQHmYR/test-utils-e42e031d31ea63d593e47d564545149a0f30b95f.zip

    no previously-included directories found matching 'example-project'
Installing collected packages: test-utils
  Running setup.py install for test-utils

    no previously-included directories found matching 'example-project'
Successfully installed test-utils
Cleaning up...
Unpacking /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-CgQad5/translate-toolkit-1.13.0.tar.bz2
  Running setup.py (path:/var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/pip-61D4bz-build/setup.py) egg_info for package from file:///var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/peep-CgQad5/translate-toolkit-1.13.0.tar.bz2
    Traceback (most recent call last):
      File "<string>", line 17, in <module>
      File "/var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/pip-61D4bz-build/setup.py", line 28, in <module>
        from sphinx.setup_command import BuildDoc
      File "/Users/sudheeshsinganamalla/Documents/pontoon/testvenv/lib/python2.7/site-packages/sphinx/setup_command.py", line 23, in <module>
        from sphinx.application import Sphinx
      File "/Users/sudheeshsinganamalla/Documents/pontoon/testvenv/lib/python2.7/site-packages/sphinx/application.py", line 26, in <module>
        from docutils.parsers.rst import convert_directive_function, \
      File "/Users/sudheeshsinganamalla/Documents/pontoon/testvenv/lib/python2.7/site-packages/docutils/parsers/rst/__init__.py", line 74, in <module>
        import docutils.statemachine
      File "/Users/sudheeshsinganamalla/Documents/pontoon/testvenv/lib/python2.7/site-packages/docutils/statemachine.py", line 113, in <module>
        from docutils import utils
      File "/Users/sudheeshsinganamalla/Documents/pontoon/testvenv/lib/python2.7/site-packages/docutils/utils/__init__.py", line 20, in <module>
        import docutils.io
      File "/Users/sudheeshsinganamalla/Documents/pontoon/testvenv/lib/python2.7/site-packages/docutils/io.py", line 18, in <module>
        from docutils.utils.error_reporting import locale_encoding, ErrorString, ErrorOutput
      File "/Users/sudheeshsinganamalla/Documents/pontoon/testvenv/lib/python2.7/site-packages/docutils/utils/error_reporting.py", line 47, in <module>
        locale_encoding = locale.getlocale()[1] or locale.getdefaultlocale()[1]
      File "/Users/sudheeshsinganamalla/Documents/pontoon/testvenv/lib/python2.7/locale.py", line 543, in getdefaultlocale
        return _parse_localename(localename)
      File "/Users/sudheeshsinganamalla/Documents/pontoon/testvenv/lib/python2.7/locale.py", line 475, in _parse_localename
        raise ValueError, 'unknown locale: %s' % localename
    ValueError: unknown locale: UTF-8
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):

  File "<string>", line 17, in <module>

  File "/var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/pip-61D4bz-build/setup.py", line 28, in <module>

    from sphinx.setup_command import BuildDoc

  File "/Users/sudheeshsinganamalla/Documents/pontoon/testvenv/lib/python2.7/site-packages/sphinx/setup_command.py", line 23, in <module>

    from sphinx.application import Sphinx

  File "/Users/sudheeshsinganamalla/Documents/pontoon/testvenv/lib/python2.7/site-packages/sphinx/application.py", line 26, in <module>

    from docutils.parsers.rst import convert_directive_function, \

  File "/Users/sudheeshsinganamalla/Documents/pontoon/testvenv/lib/python2.7/site-packages/docutils/parsers/rst/__init__.py", line 74, in <module>

    import docutils.statemachine

  File "/Users/sudheeshsinganamalla/Documents/pontoon/testvenv/lib/python2.7/site-packages/docutils/statemachine.py", line 113, in <module>

    from docutils import utils

  File "/Users/sudheeshsinganamalla/Documents/pontoon/testvenv/lib/python2.7/site-packages/docutils/utils/__init__.py", line 20, in <module>

    import docutils.io

  File "/Users/sudheeshsinganamalla/Documents/pontoon/testvenv/lib/python2.7/site-packages/docutils/io.py", line 18, in <module>

    from docutils.utils.error_reporting import locale_encoding, ErrorString, ErrorOutput

  File "/Users/sudheeshsinganamalla/Documents/pontoon/testvenv/lib/python2.7/site-packages/docutils/utils/error_reporting.py", line 47, in <module>

    locale_encoding = locale.getlocale()[1] or locale.getdefaultlocale()[1]

  File "/Users/sudheeshsinganamalla/Documents/pontoon/testvenv/lib/python2.7/locale.py", line 543, in getdefaultlocale

    return _parse_localename(localename)

  File "/Users/sudheeshsinganamalla/Documents/pontoon/testvenv/lib/python2.7/locale.py", line 475, in _parse_localename

    raise ValueError, 'unknown locale: %s' % localename

ValueError: unknown locale: UTF-8

----------------------------------------
Cleaning up...
Command python setup.py egg_info failed with error code 1 in /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/pip-61D4bz-build
Storing debug log for failure in /var/folders/sx/5cqr_xwj5c50nlc6by1431sm0000gn/T/tmpeDHUup
```

jotes commented:
Thx @sudheesh001 ! Imho, we can assume  that this issue happens always on MacosX and standard terminal app. Adding information (to the docs) about what kind of exception happens ("ValueError: unknown locale: UTF-8") would help next people to identify this issue faster. Of course, that's only my opinion, you should wait for @osmose.
Reference issue: https://code.djangoproject.com/ticket/5846

Osmose commented on docs/dev/install.rst:
```diff
@@ -53,6 +53,15 @@ Installation
 
+   .. note::
+  
+     In case you encounter a locale issue while installation, use the following
```
As @jotes mentioned, being specific about the error would be helpful here. Something like "You may encounter an error along the lines of `ValueError: blah` while running this command."

You may also want to include a link to the open Python issue: https://bugs.python.org/issue18378

Also, the fix itself needs to be explicit that it's asking the user to add those two lines to their `.bash_profile` or equivalent file.

Osmose commented:
I get the problem now, and I think this is the correct fix. Just left some feedback on the wording itself. Nice catch, and good fix, thanks!

After you've addressed the feedback, leave another comment and I'll take one last look before merging. :D

sudheesh001 commented:
@jotes and @Osmose Thank you very much for your feedback, I've addressed it with the following note.
I've rebased the commit after making these changes so that there's only one commit fixing one issue.

I hope this goes well. Requesting review.
<img width="1440" alt="localeerror" src="https://cloud.githubusercontent.com/assets/4545925/11490538/821b977e-97ff-11e5-80dd-772d9c61d6b4.png">

Osmose commented:
r+, looks great to me. Thanks for the fix!
